### PR TITLE
qa/suites/crimson-rados/rbd: Add new rbd image format api tests

### PR DESCRIPTION
**Pull request by Matan-B**
Use the second rbd image format while testing rbd api in the Crimson suite.
Currently the suite included only old format testing (`rbd_api_tests_old_format`).

https://shaman.ceph.com/builds/ceph/wip-matanb-crimson-only-new-rbd/
https://pulpito.ceph.com/matan-2023-03-09_15:12:05-crimson-rados-wip-matanb-crimson-only-new-rbd-distro-crimson-smithi/

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: qa/suites/crimson-rados/rbd/tasks/rbd_api_tests.yaml
Comment (athanatos) [inline qa/suites/crimson-rados/rbd/tasks/rbd_api_tests.yaml:20]: Missing end newline

**Review by athanatos — APPROVED** [review]

**Comment by Matan-B** [comment]
I will merge this PR after #50530 is merged. The new tests fail without it.

**Comment by Matan-B** [comment]
jenkins test make check

File: qa/suites/crimson-rados/rbd/tasks/rbd_api_tests.yaml
Comment (idryomov) [inline qa/suites/crimson-rados/rbd/tasks/rbd_api_tests.yaml:20]: Just curious, what is making you stick to `1` (just `layering` feature) here?  Have you tried enabling additional features?

FWIW the most widely used combination in the classic suite is `61`.

**Review by idryomov — COMMENTED** [review]
It would be nice to tweak the "test/librbd/test_librbd: Skip ListChildrenTiered" commit title to also say "(Crimson)".

File: qa/suites/crimson-rados/rbd/tasks/rbd_api_tests.yaml
Comment (Matan-B) [reply qa/suites/crimson-rados/rbd/tasks/rbd_api_tests.yaml:20]: Based on `test_librbd.cc` `get_features` function.
if `RBD_FEATURES` has any value set - we use new format, else we use old one
Therefore if we set `RBD_FEATURES ="1"` the new format will be tested.

This is also used in classic tests.

```
static int get_features(bool *old_format, uint64_t *features)
{
  const char *c = getenv("RBD_FEATURES");
  if (c && strlen(c) > 0) {
    stringstream ss;
    ss << c;
    ss >> *features;
    if (ss.fail())
      return -EINVAL;
    *old_format = false;
    cout << "using new format!" << std::endl;
  } else {
    *old_format = true;
    *features = 0;
    cout << "using old format" << std::endl;
  }

  return 0;
}
```

Comment (idryomov) [reply qa/suites/crimson-rados/rbd/tasks/rbd_api_tests.yaml:20]: Is the intention to test _just_ the new format or the feature set that comes with the new format?  To enable tests for what users get exposed to when they create an RBD image without specifying any additional options, `RBD_FEATURES` must be set to `61`.

**Review by idryomov — COMMENTED** [review]
LGTM but I'm wondering if this was run through teuthology after `1` -> `61` change?  That should have enabled a bunch of new tests and I'm a little surprised that it's still the case that only one test needed to be silenced due to https://tracker.ceph.com/issues/58263.

**Comment by Matan-B** [comment]
https://pulpito.ceph.com/matan-2023-03-22_13:10:56-crimson-rados-wip-matanb-crimson-only-testing-22.3-distro-crimson-smithi/

Branches included:
```
Merge branch 'wip-matanb-c-snaps-tests' into wip-matanb-crimson-only-testing-22.3                                                
Merge branch 'wip-matanb-crimson-only-order-snap' into wip-matanb-crimson-only-testing-22.3
Merge branch 'wip-matanb-c-logging-levels' into wip-matanb-crimson-only-testing-22.3 
Merge branch 'wip-matanb-c-new-rbd-api' into wip-matanb-crimson-only-testing-22.3
```

**Comment by Matan-B** [comment]
https://pulpito.ceph.com/matan-2023-03-22_13:10:56-crimson-rados-wip-matanb-crimson-only-testing-22.3-distro-crimson-smithi/7216275/
@athanatos, Should I merge this additional test although it fails atm? It will help identifying unsupported cases until they are resovled.


```
[  FAILED  ] TestLibRBD.BreakLock (5449 ms)
[  FAILED  ] TestLibRBD.QuiesceWatchError (5262 ms)
[ RUN      ] TestLibRBD.ConcurentOperations - hits an assertion
```

**Comment by idryomov** [comment]
> ```
> [  FAILED  ] TestLibRBD.BreakLock (5449 ms)
> [  FAILED  ] TestLibRBD.QuiesceWatchError (5262 ms)
> [ RUN      ] TestLibRBD.ConcurentOperations - hits an assertion
> ```
This is only what failed before `ceph_assert` terminated the whole thing.  I would suggest getting through a run with `TestLibRBD.ConcurentOperations` disabled to get the complete list of failures.

**Review by idryomov — APPROVED** [review]
Out of five "skip this test" one-line commits, three have no description and the description in the forth just duplicates a code comment.  I would suggest squashing all of them into a single commit, as it's pretty self explanatory.

File: src/test/cls_rbd/test_cls_rbd.cc
Comment (idryomov) [inline src/test/cls_rbd/test_cls_rbd.cc:1801]: It would be nice if
- the commit title reflected that it's only skipped in Crimson, e.g. "test/cls_rbd/test_cls_rbd: Skip mirror_image_status (Crimson)"
- the commit message explained the reason, e.g. "XYZ is currently not supported"

Comment (idryomov) [reply src/test/cls_rbd/test_cls_rbd.cc:1801]: > This is an attempt to establish a Crimson baseline and from there we will resolve the skipped tests.

"Establishing a baseline" is a perfectly OK thing to put in the commit message ;)

Comment (Matan-B) [reply src/test/cls_rbd/test_cls_rbd.cc:1801]: Updated.

**Comment by Matan-B** [comment]
jenkins test make check

**Comment by Matan-B** [comment]
jenkins test make check

**Review by athanatos — APPROVED** [review]
Nice!

**Comment by Matan-B** [comment]
jenkins test make check

**Comment by Matan-B** [comment]
jenkins test make check

**Pull request merged**
